lib/browser/growl.js

```diff
@@ -1,5 +1,207 @@
 'use strict';
 
-// just stub out growl
+/**
+ * Web Notifications module.
+ * @module Growl
+ */
 
-module.exports = require('../utils').noop;
+/**
+ * @summary
+ * Checks if browser notification support exists.
+ *
+ * @public
+ * @see {@link https://caniuse.com/#feat=notifications|Browser support (notifications)}
+ * @see {@link https://caniuse.com/#feat=promises|Browser support (promises)}
+ * @see {@link Mocha#growl}
+ * @see {@link Mocha#isGrowlCapable}
+ * @return {boolean} whether browser notification support exists
+ */
+exports.isCapable = function() {
+  var hasNotificationSupport = 'Notification' in window;
+  var hasPromiseSupport = typeof Promise === 'function';
+  return process.browser && hasNotificationSupport && hasPromiseSupport;
+};
+
+/**
+ * Implements browser notifications as a pseudo-reporter.
+ *
+ * @public
+ * @see {@link https://developer.mozilla.org/en-US/docs/Web/API/notification|Notification API}
+ * @see {@link https://developers.google.com/web/fundamentals/push-notifications/display-a-notification|Displaying a Notification}
+ * @see {@link Growl#isPermitted}
+ * @see {@link Mocha#_growl}
+ * @param {Runner} runner - Runner instance.
+ */
+exports.notify = function(runner) {
+  var promise = isPermitted();
+
+  /**
+   * Attempt notification.
+   */
+  var sendNotification = function() {
+    // If user hasn't responded yet... "No notification for you!" (Seinfeld)
+    Promise.race([promise, Promise.resolve(undefined)])
```

Comment:
Hey, there you have it, an actual use case for `Promise.race()`!  never seen one in the wild before.

`Promise.resolve(undefined)` is essentially the same thing as `Promise.resolve()` but with a nonempty `arguments` array.  It looks like `promise` will resolve with a boolean... maybe this should be `Promise.resolve(false)` or just `Promise.resolve()`?

not demanding this change; I'm happy to tweak it afterwards and save you the work, since you've 
 already done plenty of iteration here.

Reply:
It must **not** become `Promise.resolve(false)`, as this is the third possible state (which is used when creating the error message). Think the explicit `undefined` provides better context of _where_ the third state value comes from, but whatever.